### DISCUSSION TITLE
How should I check if a player is inside of water or lava?

### DISCUSSION THREAD

**Discussion by VendoAU** [Minestom/Minestom#643] [Help] [answered]
I've asked this question on the discord before, but I didn't really get an answer and I ended up forgetting about it.

`player.getInstance().getBlock(player.getPosition())` seems to return the correct block that the player is inside, but it does not work with water or lava.
Is this a bug or is there another way I should be doing this?

**Comment by TheMode** [comment, marked as answer]
Could you explain what you mean by "does not work"?

**Reply by VendoAU** [reply]
It returns air instead of water or lava

**Reply by TheMode** [reply]
I cannot reproduce.

**Reply by VendoAU** [reply]
Using only this code it still prints air when the player is completely submersed in water or lava
I am using jitpack with `5b91e0c622` as `d622eab048` isn't working

```
public class WaterTest {

    public static void main(String[] args) {
        final MinecraftServer minecraftServer = MinecraftServer.init();

        final InstanceManager instanceManager = MinecraftServer.getInstanceManager();
        final InstanceContainer instanceContainer = instanceManager.createInstanceContainer();
        instanceContainer.setChunkGenerator(new SuperFlatChunkGenerator());
        instanceContainer.setTime(6000);
        instanceContainer.setTimeRate(0);

        final GlobalEventHandler globalEventHandler = MinecraftServer.getGlobalEventHandler();
        globalEventHandler.addListener(PlayerLoginEvent.class, event -> {
            final Player player = event.getPlayer();

            event.setSpawningInstance(instanceContainer);
            player.setRespawnPoint(new Pos(0, 5, 0));
            player.setGameMode(GameMode.CREATIVE);
            player.setAllowFlying(true);
        });

        globalEventHandler.addListener(PlayerMoveEvent.class, event -> {
            final Player player = event.getPlayer();
            System.out.println(player.getInstance().getBlock(player.getPosition()));
        });

        minecraftServer.start("0.0.0.0", 25565);
    }
}
```

**Reply by TheMode** [reply]
You are still not explaining how to reproduce. If you are using a bucket be aware that most of it is client-predicted but not implemented on the server (you can reconnect)

**Reply by VendoAU** [reply]
I'm dumb, I was using a bucket.
I probably should have realised it was only on the client.
Thanks for your help.

**Reply by TheMode** [reply]
No problem. Would just suggest providing more information in the question next time.
